js/modules/types/attachment.js

```diff
@@ -176,10 +97,13 @@ exports._replaceUnicodeOrderOverridesSync = (attachment) => {
 exports.replaceUnicodeOrderOverrides = async attachment =>
   exports._replaceUnicodeOrderOverridesSync(attachment);
 
-// Public API
-const toVersion1 = exports.withSchemaVersion(1, autoOrientJPEG);
-const toVersion2 = exports.withSchemaVersion(2, exports.replaceUnicodeOrderOverrides);
+exports.removeSchemaVersion = (attachment) => {
+  if (!exports.isValid(attachment)) {
+    console.log('Attachment.removeSchemaVersion: Invalid input attachment:', attachment);
+    return attachment;
+  }
 
-// UpgradeStep
-exports.upgradeSchema = async attachment =>
-  toVersion2(await toVersion1(attachment));
+  const attachmentWithoutSchemaVersion = Object.assign({}, attachment);
+  delete attachmentWithoutSchemaVersion.schemaVersion;
```

Comment:
What, no, `_.omit()`? Where's the purity?? :0D

Reply:
Haha, good call! Well, purity is like ‘If a tree falls in the forest…’: If you mutate inside a function and no one is around to observe it, is the function not pure? 😉